File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/personal.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { PersonalBuscarRequest } from '../dto/request/personal-buscar.request';
import { PersonalRequest } from '../dto/request/personal.request';
import { OutResponse } from '../dto/response/out.response';
import { PersonalResponse } from '../dto/response/personal.response';

@Injectable()
export class PersonalService {

  constructor(private http: HttpClient) { }

  public listarPersonal(req: PersonalBuscarRequest): Observable<OutResponse<PersonalResponse[]>> {
    return this.http.post<OutResponse<PersonalResponse[]>>(`${environment.WsPanaderiaBackend}/personal/listarPersonal`, req);
  }

  public registrarPersonal(req: PersonalRequest): Observable<OutResponse<PersonalResponse>> {
    return this.http.post<OutResponse<PersonalResponse>>(`${environment.WsPanaderiaBackend}/personal/registrarPersonal`, req);
  }

  public modificarPersonal(req: PersonalRequest): Observable<OutResponse<PersonalResponse>> {
    return this.http.post<OutResponse<PersonalResponse>>(`${environment.WsPanaderiaBackend}/personal/modificarPersonal`, req);
  }

  public eliminarPersonal(req: PersonalRequest): Observable<OutResponse<PersonalResponse>> {
    return this.http.post<OutResponse<PersonalResponse>>(`${environment.WsPanaderiaBackend}/personal/eliminarPersonal`, req);
  }
}

File: panaderia-backend/src/main/java/com/besoft/panaderia/service/PersonalService.java
package com.besoft.panaderia.service;

import java.util.List;

import com.besoft.panaderia.dto.request.PersonalBuscarRequest;
import com.besoft.panaderia.dto.request.PersonalRequest;
import com.besoft.panaderia.dto.response.FileResponse;
import com.besoft.panaderia.dto.response.OutResponse;
import com.besoft.panaderia.dto.response.PersonalResponse;

public interface PersonalService {

	public OutResponse<List<PersonalResponse>> listarPersonal(PersonalBuscarRequest req);

	public OutResponse<PersonalResponse> registrarPersonal(PersonalRequest req);

	public OutResponse<PersonalResponse> modificarPersonal(PersonalRequest req);

	public OutResponse<PersonalResponse> eliminarPersonal(PersonalRequest req);

	public OutResponse<FileResponse> reporteXlsxListarPersonal(PersonalBuscarRequest req);
}
